src/client/api/RestClient.js

```diff
@@ -21,41 +25,66 @@ class RestClient {
     return this.peerUrl + absPath
   }
 
-  getRequest (path: string): Promise<Object> {
+  getRequest (path: string): Promise<RestResponse> {
     return this.client(this._makeUrl(path))
   }
 
-  postRequest (path: string, body: Object): Promise<Object> {
+  postRequest (path: string, body: Object | string, isJSON: boolean = true): Promise<RestResponse> {
     return this.client({
       path: this._makeUrl(path),
       method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      entity: JSON.stringify(body)
+      headers: isJSON ? { 'Content-Type': 'application/json' } : {},
+      entity: isJSON ? JSON.stringify(body) : body
     })
   }
 
   id (): Promise<string> {
-    return this.getRequest('id')
-      .then(response => response.entity)
+    return unpack(this.getRequest('id'))
   }
 
   ping (peerId: string): Promise<bool> {
-    return this.getRequest(`ping/${peerId}`)
+    return unpack(this.getRequest(`ping/${peerId}`))
       .then(response => true)
   }
 
   publish (namespace: string, statement: SimpleStatement): Promise<string> {
     console.log(`publishing ${JSON.stringify(statement)} to ${namespace}`)
-    return this.postRequest(`publish/${namespace}`, statement)
-      .then(response => response.entity)
+    return unpack(this.postRequest(`publish/${namespace}`, statement))
   }
 
   statement (statementId: string): Promise<Statement> {
-    return this.getRequest(`stmt/${statementId}`)
-      .then(response => JSON.parse(response.entity))
+    return unpack(this.getRequest(`stmt/${statementId}`), true)
+  }
+
+  getStatus (): Promise<NodeStatus> {
+    return unpack(this.getRequest('/status'))
+  }
+
+  setStatus (status: NodeStatus): Promise<NodeStatus> {
+    return unpack(
+      this.postRequest(`/status/${status}`, '', false)
+    )
   }
+
+  getDirectoryId (): Promise<string> {
+    return unpack(this.getRequest('/config/dir'))
+  }
+
+  setDirectoryId (id: string): Promise<boolean> {
+    return unpack(this.postRequest('/config/dir', id, false))
+      .then(() => true)
+  }
+}
+
+function unpack(responsePromise: Promise<RestResponse>, isJSON: boolean = false): Promise<string | Object> {
```

Comment:
This is kind of an awkward way to handle promises, though I definitely see why you did it like this. I would

* break out the success branch into its own `unpack` which is `RestResponse -> Promise<string | Object>`
* move the error formatting into a `catch` down in `getRequest`/`postRequest` (both of those funcs can be written as `return this.client({ path: this._makeUrl(path), ...args }`, right?

and overall 

* throw errors instead of manually constructing rejected promises
* `.catch` instead of using the second arg to `.then`

The motivation is to use the full power of promises: if you're manually handling and repacking errors like this then the promises are little more than glorified callbacks (I basically pretend that the second arg doesn't exist)